src/MicrosoftTeams.ts

```diff
@@ -449,7 +581,11 @@ namespace microsoftTeams
         {
             ensureInitialized(frameContexts.authentication);
 
-            sendMessage("authentication.authenticate.success", result);
+            sendMessageRequest(parentWindow, "authentication.authenticate.success", result);
+
+            setTimeout(() => {
+                window.close();
+            }, 100);
         }
```

Comment:
The timeout is actually unnecessary. I previously thought that postMessage was failing if the window got closed too quickly after it was called; however, I later on figured out that the IE/Edge issue was actually caused by window.opener getting reset in certain auth flows and had nothing to do with the window.close() call. I've removed the timeout.